src/components/Button.tsx

```diff
@@ -196,46 +199,63 @@ const Button = (
     },
   };
 
-  return (
-    <MuiButton
-      ref={ref}
-      variant={variant}
-      color={buttonColor}
-      disabled={disabled}
-      sx={{
+  const buttonProps = {
+    ref,
+    variant,
+    disabled,
+    sx: {
+      color: loading ? 'transparent !important' : 'inherit',
+      borderRadius: '3px',
+      padding: chunky ? '15px 23px' : '5px 15px',
+      zIndex: 1,
+      '&::before': {
+        content: '""',
+        height: '100%',
+        width: '0%',
+        right: 0,
+        position: 'absolute',
+        transition: 'width 500ms cubic-bezier(0.645, 0.045, 0.355, 1.000)',
+        zIndex: -1,
         borderRadius: '3px',
-        padding: chunky ? '15px 23px' : '5px 15px',
-        zIndex: 1,
+      },
+      '&:hover': {
         '&::before': {
-          content: '""',
-          height: '100%',
-          width: '0%',
-          right: 0,
-          position: 'absolute',
-          transition: 'width 500ms cubic-bezier(0.645, 0.045, 0.355, 1.000)',
-          zIndex: -1,
-          borderRadius: '3px',
+          width: '100%',
         },
-        '&:hover': {
-          '&::before': {
-            width: '100%',
-          },
-        },
-        '&.MuiButton-containedPrimary': containedPrimary,
-        '&.MuiButton-containedSecondary': containedSecondary,
-        '&.MuiButton-containedError': containedDanger,
-        '&.MuiButton-outlinedPrimary': outlinedPrimary,
-        '&.MuiButton-outlinedSecondary': outlinedSecondary,
-        '&.MuiButton-outlinedError': outlinedDanger,
-        '&.MuiButton-textPrimary': textPrimary,
-        '&.MuiButton-textSecondary': textSecondary,
-        '&.MuiButton-textError': textDanger,
-        ...sx,
-      }}
-      {...props}
+      },
+      '&.MuiButton-containedPrimary': containedPrimary,
+      '&.MuiButton-containedSecondary': containedSecondary,
+      '&.MuiButton-containedError': containedDanger,
+      '&.MuiButton-outlinedPrimary': outlinedPrimary,
+      '&.MuiButton-outlinedSecondary': outlinedSecondary,
+      '&.MuiButton-outlinedError': outlinedDanger,
+      '&.MuiButton-textPrimary': textPrimary,
+      '&.MuiButton-textSecondary': textSecondary,
+      '&.MuiButton-textError': textDanger,
+      ...sx,
+    },
+    ...props
+  }
+
+  const muiButton = (
+    <MuiButton
+      color={buttonColor}
+      {...buttonProps}
     >
       {children}
     </MuiButton>
   );
+
+  const loadingButton = (
+    <LoadingButton
+      color={buttonColor}
+      loading={loading}
+      {...buttonProps}
+    >
+      {children}
+    </LoadingButton>
+  );
+
+  return loading ? loadingButton : muiButton;
```

Comment:
I think early return would fit here nicely similarly to how the iconButton works. So something like this

```suggestion
  if (loading) {
    return (
      <LoadingButton
        color={buttonColor}
        loading={loading}
        {...buttonProps}
      >
        {children}
      </LoadingButton>
    )
  }
  
  return (
    <MuiButton
      color={buttonColor}
      {...buttonProps}
    >
      {children}
    </MuiButton>
  );
```

Reply:
And since loading will always be true when we render LoadingButton you could just pass the boolean
`loading` instead of explicitly giving it a value `loading={loading}`